### ISSUE TITLE
Key Mapper only executes the oldest known shortcut from Macrodroid shortcuts for all triggers (with different shortcuts)

### ISSUE THREAD

**Issue by arpanghosh8453** [keymapperorg/KeyMapper#1307]
**Developer TODO (don't remove)**
- [ ] create new branch. put issue number at start of name if not a very quick fix.
- [ ] write tests. put issue number in comment
- [ ] update documentation
- [ ] merge and delete branch (don't squash because want commit history to see why I made changes)

**Describe the bug**
I was hoping to use the shortcuts from macrodroid app in order to send HTTP requests for certain endpoints. There is a bug where if two or more of the shortcuts from this app are used as Key Mapper actions, only the older one will get executed. 

**To Reproduce**
1. Install macrodroid or any other shortcut launcher app
2. Create 2 or more shortcuts in that app with different names
3. Add the two different shortcuts as app shortcuts in two separate triggers in the Key Mapper app
4. Execute both of them, you will notice only the one shortcut you added first to Key Mapper will get executed in both triggers 

**Expected behavior**
The correct shortcut to get run in each trigger, instead of the oldest one key mapper knows about from the shortcut app. 

**Smartphone (please complete the following information):**
 - Device: [Samsung SM-M325F]
 - Android version: [Android 13]

**Additional context** : Related to #1222

**Comment by sds100** [comment]
Can you share your Macrodroid config or send a screenshot of the actions? I'm struggling to get the macrodroid actions to trigger at all from Key Mapper.

**Comment by arpanghosh8453** [comment]
Sure. I hope you have Macrodroid installed already. Here is what I am using this one (please remove the trailing .txt from the filename, GitHub doesn't allow .macro extension )
[Kobo_flip_left.macro.txt](https://github.com/user-attachments/files/17178991/Kobo_flip_left.macro.txt)

that macro has constrains that might now work for you (MAC specific device). You may need to remove it. Also the HTTP endpoint will not work for you as that is local. You should test the macro first on Macrodroid to see if it's working

![Kobo_flip_left](https://github.com/user-attachments/assets/e85eda17-ca42-4007-bb25-cf3ed67a5321)

So, given my macro is complicated, here is a simplified one you can test with, make another variation of it with different action. 
[Test-Macro.macro.txt](https://github.com/user-attachments/files/17178999/Test-Macro.macro.txt)

![WhatsApp Image 2024-09-29 at 16 30 08](https://github.com/user-attachments/assets/0cee4627-a6db-4c51-a847-b4d860d2d30a)

**Comment by sds100** [comment]
Fixed

**Comment by arpanghosh8453** [comment]
Thank you, any idea when this will go live in the production so I can use the updated version from the app store?

**Comment by sds100** [comment]
Can you confirm its fixed by getting a testing build as described here? https://docs.keymapper.club/contributing/#becoming-a-tester

**Comment by arpanghosh8453** [comment]
Hey @sds100, I have tested the [keymapper-2.6.3-ci.93.apk](https://cdn.discordapp.com/attachments/717532902346326088/1289922837670269032/keymapper-2.6.3-ci.93.apk?ex=66fa95c2&is=66f94442&hm=7a85bb98b06b8876089f3ecab0ec64163aa7d5e304a6f330d13dc903d8ea13db&) from the discord channel and it worked on my device as intended! Yay! Thank you so much for this amazing app and making this free and open source!

I had some trouble giving it all the permission as this is installed from the APK and there was many security checks in place that I had to bypass. That wont be an issue for the official release. Looking forward to it. Till then, I will keep using the alpha build if I don't encounter any major bug. I can now flip my Kobo e-reader pages with my Flipper zero lol!
